File: levy/python/simple.py
#!/usr/bin/env python3

from periphery import Serial

PORT_NAME = "/dev/ttyUSB1"
BAUD_RATE = 4000000
READ_TIMEOUT_SEC =  0.1

def write_byte(serial, byte):
    serial.write(bytes([byte]))

def main():

    serial = Serial(PORT_NAME, BAUD_RATE)

    write_byte(serial, 0xC0)
    write_byte(serial, 0x4A)
    write_byte(serial, 0x6A)
    write_byte(serial, 0x01)
    write_byte(serial, 0x4A)
    write_byte(serial, 0x6A)
    write_byte(serial, 0x01)
    write_byte(serial, 0x4A)
    write_byte(serial, 0x6A)
    write_byte(serial, 0x01)
    write_byte(serial, 0x4A)
    write_byte(serial, 0x6A)
    write_byte(serial, 0x01)
    write_byte(serial, 0x4A)
    write_byte(serial, 0x6A)
    write_byte(serial, 0x01)
    write_byte(serial, 0x80)

    while True:
        b = serial.read(1, READ_TIMEOUT_SEC)
        if len(b) == 0:
            break
        print("x%02X" % ord(b))

main()

File: levy/cpp/simple.cpp
#include <stdio.h>

#include "newserial.h"

int main()
{
    Serial::Begin();

    Serial::WriteByte(0xC0);
    Serial::WriteByte(0x4A);
    Serial::WriteByte(0x6A);
    Serial::WriteByte(0x01);
    Serial::WriteByte(0x4A);
    Serial::WriteByte(0x6A);
    Serial::WriteByte(0x01);
    Serial::WriteByte(0x4A);
    Serial::WriteByte(0x6A);
    Serial::WriteByte(0x01);
    Serial::WriteByte(0x4A);
    Serial::WriteByte(0x6A);
    Serial::WriteByte(0x01);
    Serial::WriteByte(0x4A);
    Serial::WriteByte(0x6A);
    Serial::WriteByte(0x01);
    Serial::WriteByte(0x80);

    while (Serial::Available()) {
        printf("x%02x\n", Serial::ReadOne());
    }

    Serial::Close();

    return 0;
}
